server/src/infra/graphql/mutation.rs

```diff
@@ -163,6 +163,48 @@ fn unpack_attributes(
     })
 }
 
+fn consolidate_attributes(
+    attributes: Vec<AttributeValue>,
+    first_name: Option<String>,
+    last_name: Option<String>,
+    avatar: Option<String>,
+) -> Vec<AttributeValue> {
+    // Prepare map with the attributes from fields, which will have lower
+    // priority than a value from a corresponding attribute already given.
+    let mut fallback_attrs: BTreeMap<String, AttributeValue> = BTreeMap::new();
+    if let Some(first_name) = first_name {
```

Comment:
It feels like you could have a for loop over `(first_name, "first_name"), (last_name, "last_name"), ...`

Reply:
Changed to that.